packages/sdk-nextjs/lib/utils/generate-id.ts

```diff
@@ -0,0 +1,42 @@
+import { customAlphabet } from "nanoid/non-secure";
+import { z } from "zod";
+
+/**
+ * Creates an ID generator with built-in validation.
+ *
+ * @param prefix - The prefix for the generated IDs (e.g., 'user', 'message')
+ * @returns An object containing ID generation and validation utilities
+ *
+ * @example
+ * const userIds = createIdGenerator('user');
```

Comment:
IMO:

The example shows `const userIds = createIdGenerator('user');`, but all the actual usage involves passing words with vowels removed as arguments. 
Aligning the example with the actual usage would enhance readability.